src/natcap/invest/hra.py

```diff
@@ -1110,153 +1095,128 @@ def _read_block(raster_path, block_info, mask_array):
-                        for prefix in ('E', 'C', 'R'):
-                            for suffix, initial_value in [
-                                    ('MIN', float('inf')), ('MAX', 0),
-                                    ('SUM', 0), ('N_PIXELS', 0)]:
-                                stats[f'{prefix}_{suffix}'] = initial_value
-
-                            for r_stat in ('R_N_HIGH', 'R_N_MEDIUM', 'R_N_LOW',
-                                           'R_N_ANY'):
-                                stats[r_stat] = 0
-
-                    for prefix, raster in [('E', e_raster),
-                                           ('C', c_raster),
-                                           ('R', r_raster)]:
-                        valid_pixels = _read_block(raster, block_info,
-                                                   subregion_mask)
-                        if valid_pixels.size == 0:
-                            # If no valid pixels, fall back to defaults.
-                            pixel_min = float('inf')
-                            pixel_max = pixel_sum = n_pixels = 0
-                        else:
-                            pixel_min = numpy.min(valid_pixels)
-                            pixel_max = numpy.max(valid_pixels)
-                            pixel_sum = numpy.sum(valid_pixels)
-                            n_pixels = valid_pixels.size
-
-                        if pixel_min < stats[f'{prefix}_MIN']:
-                            stats[f'{prefix}_MIN'] = pixel_min
-
-                        if pixel_max > stats[f'{prefix}_MAX']:
-                            stats[f'{prefix}_MAX'] = pixel_max
-
-                        stats[f'{prefix}_SUM'] += pixel_sum
-                        stats[f'{prefix}_N_PIXELS'] += n_pixels
-
-                    # the percentage of high/medium/low classification is a
-                    # per-category count.
-                    classes_block = _read_block(classes_raster, block_info,
-                                                subregion_mask)
-                    stats['R_N_HIGH'] += numpy.count_nonzero(
-                        classes_block == 3)
-                    stats['R_N_MEDIUM'] += numpy.count_nonzero(
-                        classes_block == 2)
-
-                    # Old HRA included 0-value pixels in the R_%_LOW bucket so
-                    # we do here as well.
-                    stats['R_N_LOW'] += numpy.sum(classes_block <= 1)
-                    stats['R_N_ANY'] += classes_block.size
-                    subregion_stats[subregion_id] = stats
-
-        for subregion_id, stats in subregion_stats.items():
-            # Avoid dividing by zero.  If this is 0, then the other counts must
-            # all also be 0.
-            reclassified_count = max(stats['R_N_ANY'], 1)
+                        subregion_stats[fieldname] = reduce_func([
+                            subregion_stats[fieldname],
+                            stats_under_feature[opname.lower()]])
```

Comment:
Yes, the intent is to support multiple features having the same name being treated as the same subregion.

I'm not sure I understand what you mean by accumulating them further ... since `subregion_stats_by_name` is a `collections.defaultdict()`, we're initializing the values the first time a subregion is identified so this `reduce_func` accumulation should be the only accumulation that's needed.

The only other sort of accumulation that should be needed would be if no subregion names are provided, in which case we default to all regions having the name `"Total Region"`.  In this case, the same `reduce_func` accumulation applies, we're just accumulating over all the features' stats.

I _think_ we're handling these cases as expected in the tests, too, under `test_summary_stats()`, but I could absolutely be missing something.

Could you elaborate a little on the further accumulation?

Reply:
> I'm not sure I understand what you mean by accumulating them further 

I was referring to the `zonal_stats` operation as the first accumulation/reduction -- across pixels in a feature. So I was trying to process why we would still need to be calling a a reducing function _after_ we already found the `min, max, etc` from `zonal_stats`. But you've answered that...it's the need to reduce (or accumulate, to re-use my poorly chosen term) across features with the same name.